cmake/compiler/xcc-clang/compiler_flags.cmake

```
@@ -2,10 +2,25 @@
 
 include(${ZEPHYR_BASE}/cmake/compiler/clang/compiler_flags.cmake)
 
-# Clear "nostdinc"
+# nostdinc_include contains path to llvm headers and also relative
+# path of "include-fixed".
+# Clear "nostdinc" and nostdinc_include
 set_compiler_property(PROPERTY nostdinc)
 set_compiler_property(PROPERTY nostdinc_include)
 
+# For C++ code, re-add the standard includes directory which was
+# cleared up from nostdinc_inlcude in above lines with no
+# "include-fixed" this time"
+if(CONFIG_CPP)
+  execute_process(
+    COMMAND ${CMAKE_C_COMPILER} --print-file-name=include/stddef.h
+    OUTPUT_VARIABLE _OUTPUT
+    )
```

Maybe you want this, it's surprisingly not the default:
```suggestion
    COMMAND_ERROR_IS_FATAL ANY
    )
```

---

Added this.